Quantum-Control-Applications/Superconducting/Single Fixed Transmon/qubit_spec.py

```diff
@@ -56,58 +58,71 @@
 #####################################
 qmm = QuantumMachinesManager(qop_ip)
 
-###############
-# Run Program #
-###############
-qm = qmm.open_qm(config)
+###########################
+# Run or Simulate Program #
+###########################
 
-job = qm.execute(qubit_spec)
+simulate = True
 
-res_handles = job.result_handles
-I_handle = res_handles.get("I")
-Q_handle = res_handles.get("Q")
-iteration_handle = res_handles.get("iteration")
-I_handle.wait_for_values(1)
-Q_handle.wait_for_values(1)
-iteration_handle.wait_for_values(1)
-next_percent = 0.1  # First time print 10%
+if simulate:
+    simulation_config = SimulationConfig(duration=1000)
+    job = qmm.simulate(config, qubit_spec, simulation_config)
+    job.get_simulated_samples().con1.plot()
 
+else:
+    qm = qmm.open_qm(config)
 
-def on_close(event):
-    event.canvas.stop_event_loop()
-    job.halt()
+    job = qm.execute(qubit_spec)
 
+    res_handles = job.result_handles
+    I_handle = res_handles.get("I")
+    Q_handle = res_handles.get("Q")
+    iteration_handle = res_handles.get("iteration")
+    I_handle.wait_for_values(1)
+    Q_handle.wait_for_values(1)
+    iteration_handle.wait_for_values(1)
+    next_percent = 0.1  # First time print 10%
 
-f = plt.figure()
-f.canvas.mpl_connect("close_event", on_close)
-print("Progress =", end=" ")
+    def on_close(event):
+        event.canvas.stop_event_loop()
+        job.halt()
 
-while res_handles.is_processing():
-    plt.cla()
-    I = I_handle.fetch_all()
-    Q = Q_handle.fetch_all()
-    iteration = iteration_handle.fetch_all()
-    if iteration / n_avg > next_percent:
-        percent = 10 * round(iteration / n_avg * 10)  # Round to nearest 10%
-        print(f"{percent}%", end=" ")
-        next_percent = percent / 100 + 0.1  # Print every 10%
+    f = plt.figure()
+    f.canvas.mpl_connect("close_event", on_close)
+    print("Progress =", end=" ")
 
-    plt.plot(freqs, np.sqrt(I**2 + Q**2), ".")
-    # plt.plot(freqs + qubit_LO, np.sqrt(I**2 + Q**2), '.')
+    while res_handles.is_processing():
+        plt.cla()
+        I = I_handle.fetch_all()
+        Q = Q_handle.fetch_all()
+        iteration = iteration_handle.fetch_all()
+        # If we want to plot the phase
+        phase = signal.detrend(np.unwrap(np.angle(I + 1j * Q)))
```

Comment:
qubit spect don't need the `signal.detrend` because the readout frequency is constant (so there is no linear trend)